### PR TITLE
feat: :sparkles: implement setting to store scopes in user settings i…

### PR DESCRIPTION
**Pull request by Hobart2967**
…nstead of workspace settings

https://github.com/vivaxy/vscode-conventional-commits/issues/335

### CODE REVIEW

**Review by vivaxy — CHANGES_REQUESTED** [review]
Thanks for the PR.
I still have some questions left.

File: package.nls.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "extension.commands.resetGlobalState": "Conventional Commits: Reset Global State",
   "extension.commands.showNewVersionNotes": "Conventional Commits: Show Version Notes",
   "extension.configuration.autoCommit.markdownDescription": "Control whether the extension should commit files after: forming the message or closing the editor tab.\n\nWhen `#git.enableSmartCommit#` enabled and `#git.smartCommitChanges#` was set to `all`, It allows to commit all changes when there are no staged changes.\n\nAnd set `#git.postCommitCommand#` to `sync` to run `git.sync` after commit.",
+  "extension.configuration.storeScopesGlobally.markdownDescription": "Control whether the extension should store the defined scopes within your workspace settings or your user settings.",
```
[the remark this answers is not in the document]
Comment (Hobart2967) [reply package.nls.json:6]: Done :) Please also have a look at the corresponding chinese version. As I do not speak or can write it, I'd appreciate your help :)

File: src/lib/prompts/prompt-types.ts [hunk 2]
```diff
@@ -148,6 +150,7 @@ export type ConfigurableQuickPickOptions = {
   newItemWithoutSetting: Item;
   addNoneOption: boolean;
   validate?: (value: string) => string | undefined;
+  storeGlobal: boolean | undefined;
```
[the remark this answers is not in the document]
Comment (Hobart2967) [reply src/lib/prompts/prompt-types.ts:153]: Good point!

File: package.nls.zh-cn.json [hunk 3]
```diff
@@ -2,6 +2,7 @@
   "extension.name": "约定式提交",
   "extension.commands.resetGlobalState": "约定式提交: 重置全局状态",
   "extension.commands.showNewVersionNotes": "约定式提交: 显示版本提示",
+  "extension.configuration.storeScopesGlobally.markdownDescription": "Control whether the extension should store the defined scopes within your workspace settings or your user settings.",
```
[the remark this answers is not in the document]
Comment (Hobart2967) [reply package.nls.zh-cn.json]: Done!

File: src/lib/prompts/prompt-types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hobart2967) [reply src/lib/prompts/prompt-types.ts]: Done!

File: src/lib/commitlint.ts [hunk 4]
```diff
@@ -18,14 +19,15 @@ class Commitlint {
         output.info('Load commitlint configuration successfully.');
         return rules;
       } catch (e) {
+        const error = e as Error;
```
[the remark this answers is not in the document]
Comment (Hobart2967) [reply src/lib/commitlint.ts]: Done!

**Review by vivaxy — APPROVED** [review]

**Comment by vivaxy** [comment]
@Hobart2967 Thanks for your PR. I'll try to check and release the extension soon.

**Pull request merged**